api/metrics-api-jaxrs-1.1/src/main/java/org/hawkular/metrics/api/jaxrs/filter/CorsFilter.java

```diff
@@ -56,28 +72,50 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         final HttpServletRequest httpRequest = (HttpServletRequest) request;
         final HttpServletResponse httpResponse = (HttpServletResponse) response;
 
-        String origin = "*";
+        //NOT a CORS request
+        if (httpRequest.getHeader(ORIGIN) == null) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String requestOrigin = null;
         Enumeration<String> originHeaders = httpRequest.getHeaders(ORIGIN);
         if (originHeaders != null && originHeaders.hasMoreElements()) {
             String originHeaderValue = originHeaders.nextElement();
             if (originHeaderValue != null && !originHeaderValue.equals("null")) {
-                origin = originHeaderValue.split(",")[0];
+                requestOrigin = originHeaderValue.split(",")[0];
             }
         }
-        httpResponse.addHeader(ACCESS_CONTROL_ALLOW_ORIGIN, origin);
 
-        httpResponse.addHeader(ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
-        httpResponse.addHeader(ACCESS_CONTROL_ALLOW_METHODS, DEFAULT_CORS_ACCESS_CONTROL_ALLOW_METHODS);
-        httpResponse.addHeader(ACCESS_CONTROL_MAX_AGE, (72 * 60 * 60) + "");
-        httpResponse.addHeader(ACCESS_CONTROL_ALLOW_HEADERS, DEFAULT_CORS_ACCESS_CONTROL_ALLOW_HEADERS);
+        if (allowAnyOrigin || OriginValidation.isAllowedOrigin(requestOrigin, allowedOrigins)) {
+            httpResponse.addHeader(ACCESS_CONTROL_ALLOW_ORIGIN, requestOrigin);
+            httpResponse.addHeader(ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+            httpResponse.addHeader(ACCESS_CONTROL_ALLOW_METHODS, DEFAULT_CORS_ACCESS_CONTROL_ALLOW_METHODS);
+            httpResponse.addHeader(ACCESS_CONTROL_MAX_AGE, (72 * 60 * 60) + "");
+            httpResponse.addHeader(ACCESS_CONTROL_ALLOW_HEADERS, DEFAULT_CORS_ACCESS_CONTROL_ALLOW_HEADERS);
 
-        if (!isPreflightRequest((HttpServletRequest) request)) {
-            chain.doFilter(request, response);
+            if (!isPreflightRequest((HttpServletRequest) request)) {
+                chain.doFilter(request, response);
+            }
+        } else {
+            httpResponse.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
I haven't found in the spec a reference to that status code. Is that something common? Or did I miss the reference in the spec?